### PR TITLE
Document SVGClipPathElement.(clipPathUnits|transform)

### PR DESCRIPTION
**Pull request by teoli2003**
### Description

Create docs for `SVGClipPathElement.clipPathUnits` and `SVGClipPathElement.transform`

### Motivation

This is part of openwebdocs/project#152

### Additional details

You fall into at least one rabbit hole when you touch the SVG docs. In this case, I had to:
- Fix `SVGClipPathElement` that was incomplete.
- Fix `SVGUnitTypes` that was old school.
- Fix `SVGAnimatedEnumeration` that was even older school.
- Add `SVGAnimatedEnumeration.baseVal` and `animVal`.

Having practical examples for the latest was not trivial (as animated enumeration is not very interesting to demonstrate).

I didn't fall further into the rabbit hole as I limited myself to what was needed to avoid flaws in these files.

### Related issues and pull requests

_None._

### CODE REVIEW

File: files/en-us/web/api/svganimatedenumeration/animval/index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: SVGAnimatedEnumeration.animVal
+slug: Web/API/SVGAnimatedEnumeration/animVal
+page-type: web-api-instance-property
+browser-compat: api.SVGAnimatedEnumeration.animVal
+---
+
+{{APIRef("SVG")}}
+
+The **`animVal`** property of the {{domxref("SVGAnimatedEnumeration")}} interface contains the current value of a SVG enumeration. If there is no animation, it is the value as the {{domxref("SVGAnimatedEnumeration.baseVal", "baseVal")}}.
```
Comment (dawei-wang) [inline files/en-us/web/api/svganimatedenumeration/animval/index.md:10]:
```suggestion
The **`animVal`** property of the {{domxref("SVGAnimatedEnumeration")}} interface contains the current value of a SVG enumeration. If there is no animation, it is the same value as the {{domxref("SVGAnimatedEnumeration.baseVal", "baseVal")}}.
```

File: files/en-us/web/api/svganimatedenumeration/index.md [hunk 2]
```diff
@@ -7,78 +7,46 @@ browser-compat: api.SVGAnimatedEnumeration
...
-        attribute or property is not currently being animated, contains the same
-        value as <code>baseVal</code>.
-      </td>
-    </tr>
-  </tbody>
-</table>
+- {{domxref("SVGAnimatedEnumeration.baseVal", "baseVal")}}
+  - : An integer that is the base value of the given attribute before applying any animations.
+- {{domxref("SVGAnimatedEnumeration.animVal", "animVal")}}
+  - : If the given attribute or property is being animated, contains thecurrent animated value of the attribute or property. If the given attribute or property is not currently being animated, contains the same value as `baseVal`.
```
Comment (dawei-wang) [inline files/en-us/web/api/svganimatedenumeration/index.md:17]:
```suggestion
  - : If the given attribute or property is being animated, it contains the current animated value of the attribute or property. If the given attribute or property is not currently being animated, it contains the same value as `baseVal`.
```

File: files/en-us/web/api/svgclippathelement/clippathunits/index.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+---
+title: SVGClipPathElement.clipPathUnits
+slug: Web/API/SVGClipPathElement/clipPathUnits
+page-type: web-api-instance-property
+browser-compat: api.SVGClipPathElement.clipPathUnits
+---
+
+{{APIRef("SVG")}}
+
+The read-only **`clipPathUnits`** property of the {{domxref("SVGClipPathElement")}} interface reflects the {{SVGAttr("clipPathUnits")}} attribute of a {{SVGElement("clipPath")}} element and by that defines the coordinate system to use for the content of the element.
```
Comment (dawei-wang) [inline files/en-us/web/api/svgclippathelement/clippathunits/index.md:10]:
```suggestion
The read-only **`clipPathUnits`** property of the {{domxref("SVGClipPathElement")}} interface reflects the {{SVGAttr("clipPathUnits")}} attribute of a {{SVGElement("clipPath")}} element which defines the coordinate system to use for the content of the element.
```

File: files/en-us/web/api/svgclippathelement/clippathunits/index.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+> **Note:** Although this property is read-only, it is merely a container for two values you can modify, {{domxref("SVGAnimatedEnumeration.baseVal", "baseVal")}} and {{domxref("SVGAnimatedEnumeration.animVal", "animVal")}}.
+
+## Value
+
+An {{domxref("SVGAnimatedEnumeration")}} representing the coordinate system. The possible values are the one defined in the {{domxref("SVGUnitTypes")}} interface:
```
Comment (dawei-wang) [inline files/en-us/web/api/svgclippathelement/clippathunits/index.md:16]:
```suggestion
An {{domxref("SVGAnimatedEnumeration")}} representing the coordinate system. The possible values are defined in the {{domxref("SVGUnitTypes")}} interface:
```

File: files/en-us/web/api/svgclippathelement/clippathunits/index.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+- `0` (`SVG_UNIT_TYPE_UNKWOWN`)
+  - : The type is not one of the predefined type.
+- `1` (`SVG_UNIT_TYPE_USERSPACEONUSE`)
+  - : Corresponding to a value of `userSpaceOnUse` for the {{SVGAttr("clipPathUnits")}}attribute and means that all coordinates inside the element refer to the user coordinate system as defined when the clipping path was created. It is the default value.
```
Comment (dawei-wang) [inline files/en-us/web/api/svgclippathelement/clippathunits/index.md:21]:
```suggestion
  - : Corresponding to a value of `userSpaceOnUse` for the {{SVGAttr("clipPathUnits")}} attribute and means that all coordinates inside the element refer to the user coordinate system as defined when the clipping path was created. It is the default value.
```

File: files/en-us/web/api/svgclippathelement/transform/index.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
+---
+title: SVGClipPathElement.transform
+slug: Web/API/SVGClipPathElement/transform
+page-type: web-api-instance-property
+browser-compat: api.SVGClipPathElement.transform
+---
+
+{{APIRef("SVG")}}
+
+The read-only **`transform`** property of the {{domxref("SVGClipPathElement")}} interface reflects the {{SVGAttr("transform")}} attribute of a {{SVGElement("clipPath")}} element, that is a list of transformation applied to the element.
```
Comment (dawei-wang) [inline files/en-us/web/api/svgclippathelement/transform/index.md:10]:
```suggestion
The read-only **`transform`** property of the {{domxref("SVGClipPathElement")}} interface reflects the {{SVGAttr("transform")}} attribute of a {{SVGElement("clipPath")}} element, that is a list of transformations applied to the element.
```

**Review by dawei-wang — COMMENTED** [review]
A few grammar edits

File: files/en-us/web/api/svganimatedenumeration/animval/index.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
+---
+title: SVGAnimatedEnumeration.animVal
+slug: Web/API/SVGAnimatedEnumeration/animVal
+page-type: web-api-instance-property
+browser-compat: api.SVGAnimatedEnumeration.animVal
+---
+
+{{APIRef("SVG")}}
+
+The **`animVal`** property of the {{domxref("SVGAnimatedEnumeration")}} interface contains the current value of a SVG enumeration. If there is no animation, it is the same value as the {{domxref("SVGAnimatedEnumeration.baseVal", "baseVal")}}.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svganimatedenumeration/animval/index.md:10]:
```suggestion
The **`animVal`** property of the {{domxref("SVGAnimatedEnumeration")}} interface contains the current value of an SVG enumeration. If there is no animation, it is the same value as the {{domxref("SVGAnimatedEnumeration.baseVal", "baseVal")}}.
```

File: files/en-us/web/api/svganimatedenumeration/baseval/index.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
+---
+title: SVGAnimatedEnumeration.baseVal
+slug: Web/API/SVGAnimatedEnumeration/baseVal
+page-type: web-api-instance-property
+browser-compat: api.SVGAnimatedEnumeration.baseVal
+---
+
+{{APIRef("SVG")}}
+
+The **`baseVal`** property of the {{domxref("SVGAnimatedEnumeration")}} interface contains the initial value of a SVG enumeration.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svganimatedenumeration/baseval/index.md:10]:
```suggestion
The **`baseVal`** property of the {{domxref("SVGAnimatedEnumeration")}} interface contains the initial value of an SVG enumeration.
```

File: files/en-us/web/api/svganimatedenumeration/baseval/index.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+## Value
+
+An integer containing the initial value of the enumeration
+
+## Examples
+
+Considering this snippet with a {{SVGElement("clipPath")}} element. Its {{SVGAttr("clipPathUnits")}} is associated with a {{domxref("SVGAnimatedEnumeration")}} object.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svganimatedenumeration/baseval/index.md:18]:
```suggestion
Considering this snippet with a {{SVGElement("clipPath")}} element: Its {{SVGAttr("clipPathUnits")}} is associated with a {{domxref("SVGAnimatedEnumeration")}} object.
```

File: files/en-us/web/api/svganimatedenumeration/index.md [hunk 10]
```diff
@@ -7,78 +7,46 @@ browser-compat: api.SVGAnimatedEnumeration
...
-      <td>
-        <a
-          href="https://www.w3.org/TR/SVG11/types.html#InterfaceSVGAnimatedEnumeration"
-          >SVG 1.1 (2nd Edition)</a
-        >
-      </td>
-    </tr>
-  </tbody>
-</table>
+The **`SVGAnimatedEnumeration`** interface descrobes attributes value which are constants from a particular enumeration and which can be animated.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svganimatedenumeration/index.md:10]:
```suggestion
The **`SVGAnimatedEnumeration`** interface describes attribute values which are constants from a particular enumeration and which can be animated.
```

File: files/en-us/web/api/svganimatedenumeration/index.md [hunk 11]
```diff
@@ -7,78 +7,46 @@ browser-compat: api.SVGAnimatedEnumeration
...
+- {{domxref("SVGAnimatedEnumeration.animVal", "animVal")}}
+  - : If the given attribute or property is being animated, it contains the current animated value of the attribute or property. If the given attribute or property is not currently being animated, it contains the same value as `baseVal`.
 
 ## Instance methods
 
 The `SVGAnimatedEnumeration` interface do not provide any specific methods.
 
+## Examples
+
+Considering this snippet with a {{SVGElement("clipPath")}} element. Its {{SVGAttr("clipPathUnits")}} is associated with a `SVGAnimatedEnumeration` object.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svganimatedenumeration/index.md:25]:
```suggestion
Considering this snippet with a {{SVGElement("clipPath")}} element: Its {{SVGAttr("clipPathUnits")}} is associated with a `SVGAnimatedEnumeration` object.
```

File: files/en-us/web/api/svgclippathelement/clippathunits/index.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+> **Note:** Although this property is read-only, it is merely a container for two values you can modify, {{domxref("SVGAnimatedEnumeration.baseVal", "baseVal")}} and {{domxref("SVGAnimatedEnumeration.animVal", "animVal")}}.
+
+## Value
+
+An {{domxref("SVGAnimatedEnumeration")}} representing the coordinate system. The possible values are defined in the {{domxref("SVGUnitTypes")}} interface:
+
+- `0` (`SVG_UNIT_TYPE_UNKWOWN`)
+  - : The type is not one of the predefined type.
+- `1` (`SVG_UNIT_TYPE_USERSPACEONUSE`)
+  - : Corresponding to a value of `userSpaceOnUse` for the {{SVGAttr("clipPathUnits")}} attribute and means that all coordinates inside the element refer to the user coordinate system as defined when the clipping path was created. It is the default value.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svgclippathelement/clippathunits/index.md:21]:
```suggestion
  - : Corresponds to a value of `userSpaceOnUse` for the {{SVGAttr("clipPathUnits")}} attribute and means that all coordinates inside the element refer to the user coordinate system as defined when the clipping path was created. It is the default value.
```

File: files/en-us/web/api/svgclippathelement/clippathunits/index.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+- `2` (`SVG_UNIT_TYPE_OBJECTBOUNDINGBOX`)
+  - : Corresponding to a value of `objectBoundingBox` for the attribute and means that all coordinates inside the element are relative to the bounding box of the element the clipping path is applied to. It means that the origin of the coordinate system is the top left corner of the object bounding box and the width and height of the object bounding box are considered to have a length of 1 unit value.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svgclippathelement/clippathunits/index.md:23]:
```suggestion
  - : Corresponds to a value of `objectBoundingBox` for the attribute and means that all coordinates inside the element are relative to the bounding box of the element the clipping path is applied to. It means that the origin of the coordinate system is the top left corner of the object bounding box and the width and height of the object bounding box are considered to have a length of 1 unit value.
```

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**